extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -1294,21 +1304,21 @@ export class Bind {
 
   /**
    * Print out the current state in the console.
+   * @param {string=} opt_expression
    * @private
    */
-  printState_() {
-    const seen = [];
-    const s = JSON.stringify(this.state_, (key, value) => {
-      if (isObject(value)) {
-        if (seen.includes(value)) {
-          return '[Circular]';
-        } else {
-          seen.push(value);
-        }
+  printState_(opt_expression) {
+    if (opt_expression) {
+      if (typeof opt_expression !== 'string') {
+        user().error(TAG, 'Invalid JSON expression. Example: ' +
+            'AMP.printState("foo.bar") to print current value of "foo.bar".');
+        return;
```

Comment:
no need for return, it is never reached as an error is thrown. You actually commented on a similar change I made 😄

Reply:
You need to return for `error()` but not for `assert()`. Only the latter throws a JS error.